marketplacetests/marketplace/app.py

```diff
@@ -66,7 +66,9 @@ def offline_message_text(self):
 
     def wait_for_notification_message(self, message):
         element = self.marionette.find_element(*self._notification_locator)
-        Wait(self.marionette).until(lambda m: element.text == message)
+        Wait(self.marionette).until(lambda m: element.text == message,
+                                    message='Notification not found. Expected: %s | Found: %s'
+                                            % (message, element.text))
```

Comment:
The mistake I made (I think), as I commented below, was expecting that `.text` would tell me what is inside the element, regardless of whether it is displayed or not. I want to make sure my understanding of the behaviour above is correct though. I assumed that `element.text` would be continually updated through the life of the `Wait`, not that it would remain at whatever value it was when entering the loop. Is that correct? I realize that it's possible that the value of `element.text` would be different at the end, when I output the message, so you are right that logging before checking would make more sense.